From 8e0d3dc90ab7bafc59b86e9a66d08d1e2b6f3e0e Mon Sep 17 00:00:00 2001
From: "Z. Liu" <zhixu.liu@gmail.com>
Date: Tue, 15 Apr 2025 21:48:10 +0800
Subject: [PATCH] CMakeLists.txt: fix opentelemetry library

otherwise, cmake failed with:

> CMake Error at src/CMakeLists.txt:443 (target_link_libraries):
>   The link interface of target "jaeger_base" contains:
>
>     opentelemetry::libopentelemetry
>
>   but the target was not found.  Possible reasons include:
>
>     * There is a typo in the target name.
>     * A find_package call is missing for an IMPORTED target.
>     * An ALIAS target is missing.

See also: https://bugs.gentoo.org/930064

Signed-off-by: Z. Liu <zhixu.liu@gmail.com>
---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 801086e1268a1..5d30cde9790d4 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -330,7 +330,7 @@ if(WITH_JAEGER)
   include(BuildOpentelemetry)
   build_opentelemetry()
   add_library(jaeger_base INTERFACE)
-  target_link_libraries(jaeger_base INTERFACE opentelemetry::libopentelemetry
+  target_link_libraries(jaeger_base INTERFACE ${OPENTELEMETRY_CPP_LIBRARIES}
     thrift::libthrift)
 endif()
 
--- a/src/CMakeLists.txt	2025-09-17 07:24:59.548258925 -0000
+++ b/src/CMakeLists.txt	2025-09-17 07:28:31.449345268 -0000
@@ -319,6 +319,7 @@
 
 if(WITH_JAEGER)
   find_package(thrift 0.13.0 REQUIRED)
+  find_package(prometheus-cpp REQUIRED)
   find_package(opentelemetry-cpp REQUIRED)
   set(jaeger_base
 	 opentelemetry-cpp::common
